src/monster.cpp

```diff
@@ -1637,6 +1641,12 @@ void monster::process_effects()
         }
     }
 
+    if( has_effect("shrieking")){
+        sounds::sound(pos(), 120, _("a piercing wail!"), true);
+        // Lose 40% of our maximum speed while shrieking
+        set_speed_base( .6 * type->speed);
```

Comment:
Did you test that? Something tells me this won't wear off when the status is removed. Plus, it could speed up reviving zombies.